### ISSUE TITLE
Load config files once at start rather than multiple times

### ISSUE THREAD

**Issue by zinsho** [bhnn/hunter-sim#20]
Right now the config yamls for hunters are being loaded in 3 locations:

1. `hunter_sym.py` when running it as a comparison to ensure both hunters are the same
2. `hunters.py` to load the hunter and ensure all the keys are properly created
3. `sim.py` to ensure a valid hunter class is selected.

It is also being passed into the threaded simulation so each thread opens the file to read.

Opening it once during setup and creating Hunter Object(s) for the passed in configuration(s) would allow all the validation to be done right away and also allow the object to be passed to the threads in memory rather than talking to the filesystem again.

This also tied into my experiments in #19 where I wanted to just pass in a Hunter Object/payload and it kept requiring a file instead.

**Comment by bhnn** [comment]
This is a relic from earlier days as a workaround to avoid having to deep-copy the initial hunter object all the time, so I just passed the path. With the way the `SimulationManager` works now and especially with multi-processing, it's not in its best state. v0.5 will change that to doing a majority of the verification work in `hunter_sim.py` and then only pass on a build config dictionary object to create new hunter instances from.

**Comment by zinsho** [comment]
I figured it was something of the sort.

Hopefully whatever is built is such that a theoretical `api.py` file could leverage the same validation based on a JSON payload and pass it off to the sim (this might mean the validation has to be outside `hunter_sim.py` so that it becomes common code).  I've had to dabble with `fastpi` and `pydantic` for some other projects so the thought of turning this sim into a dockerfile with a frontend (or being able to host it somewhere) would certainly help adoption by people who can't/won't install and run python locally.
